scalding-core/src/main/scala/com/twitter/scalding/TypedDelimited.scala

```diff
@@ -47,15 +48,15 @@ trait TypedSeperatedFile extends Serializable {
 /**
  * Typed tab separated values file
  */
-object TypedTsv extends TypedSeperatedFile {
```

Comment:
You won't have the type information there. You probably want to take the
implicit as is but maybe copy it with fields changed.

On Thursday, July 9, 2015, Sidharth Kapur <notifications@github.com> wrote:

> In scalding-core/src/main/scala/com/twitter/scalding/TypedDelimited.scala
> <https://github.com/twitter/scalding/pull/1356#discussion_r34321088>:
>
> > @@ -47,15 +48,15 @@ trait TypedSeperatedFile extends Serializable {
> >  /**
> >   * Typed tab separated values file
> >   */
> > -object TypedTsv extends TypedSeperatedFile {
>
> So I'm thinking we could add a new macro (not implicit) in
> TypeDescriptorProviderImpl which is like caseClassTypeDescriptorCommonImpl,
> but which takes fields as a parameter and uses that value of fields,
> instead of using the helper macro.
>
> We would then call this macro from the apply methods of
> TypedSeparatedFile which specify a fields parameter.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twitter/scalding/pull/1356/files#r34321088>.
>